synapse/handlers/space_summary.py

```diff
@@ -149,20 +200,23 @@ async def federation_space_summary(
 
         while room_queue and len(rooms_result) < MAX_ROOMS:
             room_id = room_queue.popleft()
+            if room_id in processed_rooms:
+                # already done this room
+                continue
```

Comment:
This would happen because we now don't check below whether the edge is processed already while extending the queue?

Reply:
it can happen because we might find two edges pointing to the same room. I think this was actually a bug in the original impl, I just hadn't found it yet.

Example: A has two children, B and C. C is also a child of B.  First we process A, and add the discovered children (B and C) to the queue. We pop B off the queue, and add *its* children (C) to the queue. C is now in the queue twice. Without this line, we end up processing it twice.